### PR TITLE
Feat/mawqit for tv take the tv language as the app language by default

### PR DESCRIPTION
**Pull request by YassinNouh21**
---

📝 **Summary**
---
This PR introduces a key feature aimed at enhancing user experience by automatically setting the Mawaqit TV app's default language to match the TV's current language setting. 

**This PR is for issue:** #984 

**Description**
---
Leveraging the TV's language setting as the app's default language, we reduce the friction new users might face and cater to a global audience with minimal effort required from the user's end.

Key changes include:

- Automatic detection of the TV's current language setting during the app's initial setup phase.
- Dynamic adjustment of the app's language setting to align with the TV's language, ensuring immediate familiarity and ease of use for the user.
- Thoughtful handling of scenarios where the TV's language is not supported by the app, defaulting to English while providing users the option to choose their preferred language manually.

**Tests**
---
🧪 **Use case 1: Language is available in the app's localization**
---

image

---
🧪 **Use case 2: Language is not available in the app's localization**
---

image3

---
🧪 **Use case 3: Exception handling during data fetching**
--- 

image2

---
**Checklist:**
---
- [x] **Coding Standards:** I have reviewed my code to ensure it follows the project's coding standards.
- [x] **Testing:** I have thoroughly tested the changes across the described use cases and confirmed they work as expected.
- [x] **Merge Conflicts:** I have resolved any merge conflicts with the latest main/development branch.
- [x] **Branch Status:** The branch is up-to-date with the target branch (main/development).

---

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -59,7 +59,7 @@ dependencies:
   cached_network_image: ^3.2.3
   xml_parser: ^1.0.0
 
-  share_plus: ^6.3.1
+  share_plus: ^7.2.2
```
Comment (ibrahim-zehhaf-mawaqit) [inline pubspec.yaml:62]: to revert

File: lib/src/domain/usecase/on_boarding_usecase.dart [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  final DeviceInfoRepository deviceInfoRepository;
+
+  /// shared preference instance
+  OnBoardingUseCase({
+    required this.deviceInfoRepository,
+  });
+
+  Future<DeviceInfoModel> getDeviceInfo() async {
+    final deviceInfoModel = await deviceInfoRepository.getAllDeviceInfo();
+    return deviceInfoModel;
```
Comment (ghassenbenzahra123) [inline lib/src/domain/usecase/on_boarding_usecase.dart:19]: directly  ```return await deviceInfoRepository.getAllDeviceInfo();```

File: pubspec.yaml [hunk 3]
```diff
@@ -72,7 +72,7 @@ dependencies:
   device_info_plus: ^9.0.2
   android_intent_plus: ^4.0.0
   global_configuration: ^2.0.0-nullsafety.1
-  wakelock: ^0.6.2
```
Comment (ghassenbenzahra123) [inline pubspec.yaml:75]: to revert also please

**Review by ghassenbenzahra123 — CHANGES_REQUESTED** [review]
Good work Yassin! some minor updates.

**Review by ghassenbenzahra123 — APPROVED** [review]
LGTM! good work Yassin

**Comment by ibrahim-zehhaf-mawaqit** [comment]
I tested the issue, please keep the language screen in the on boarding, just select the default language, don't remove the screen

**Pull request merged**